pkg/apis/apps/validation/validation.go

```diff
@@ -132,3 +132,39 @@ func ValidateStatefulSetStatusUpdate(statefulSet, oldStatefulSet *apps.StatefulS
 	// TODO: Validate status.
 	return allErrs
 }
+
+// ValidateControllerRevisionName can be used to check whether the given ControllerRevision name is valid.
+// Prefix indicates this name will be used as part of generation, in which case
+// trailing dashes are allowed.
+var ValidateControllerRevisionName = apivalidation.NameIsDNSSubdomain
+
+// ValidateControllerRevision collects errors for the fields of state and returns those errors as an ErrorList. If the
+// returned list is empty, state is valid. Validation is performed to ensure that state is a valid ObjectMeta, its name
+// is valid, and that it doesn't exceed the MaxControllerRevisionSize.
+func ValidateControllerRevision(revision *apps.ControllerRevision) field.ErrorList {
+	errs := field.ErrorList{}
+
+	errs = append(errs, apivalidation.ValidateObjectMeta(&revision.ObjectMeta, true, ValidateControllerRevisionName, field.NewPath("metadata"))...)
+	if revision.Data == nil {
+		errs = append(errs, field.Required(field.NewPath("data"), "data is mandatory"))
+	}
+	errs = append(errs, apivalidation.ValidateNonnegativeField(revision.Revision, field.NewPath("revision"))...)
+	return errs
+}
+
+// ValidateControllerRevisionUpdate collects errors pertaining to the mutation of an ControllerRevision Object. If the
+// returned ErrorList is empty the update operation is valid. Any mutation to the ControllerRevision's Data or Revision
+// is considered to be invalid.
+func ValidateControllerRevisionUpdate(newHistory, oldHistory *apps.ControllerRevision) field.ErrorList {
+	errs := field.ErrorList{}
+
+	errs = append(errs, apivalidation.ValidateObjectMetaUpdate(&newHistory.ObjectMeta, &oldHistory.ObjectMeta, field.NewPath("metadata"))...)
+	errs = append(errs, ValidateControllerRevision(newHistory)...)
+	if newHistory.Revision != oldHistory.Revision {
+		errs = append(errs, field.Forbidden(field.NewPath("revision"), "revision can't be updated"))
+	}
```

Comment:
You convinced me of the use case for mutable Revision, and having the data be immutable is strong enough for StatefulSet's use case. I'm going to update the PR to make it `.Revision` mutable.